src/install_rules.ml

```diff
@@ -226,90 +223,41 @@ module Gen(P : Params) = struct
     | Default -> true
     | Opam _  -> false
 
-  let bin_entries, other_stanzas, install_paths_per_package =
```

Comment:
Sorry, I didn't read this PR in detail initially. This needed to be done at the toplevel of this module, in order for the rules producing symlinks in `_build/install/default/bin` to be setup ASAP. I can't find where this code was moved

Reply:
Ah I see. Where do we depend on the rules being setup before anything else?

Reply:
In `Artifacts.create`, we simply list `_build/install/default/bin` to find out the list of public binaries. This avoids duplicating the logic to interpret `install` stanzas. However, this requires the install rules to be setup at this point. The install rules for libraries requires to know the list of modules, which is done via `Dir_contents` which interprets stanza like `rule` or `copy_files`. Interpreting these stanzas requires expanding percent forms and so requires the `Artifacts.t`. Given that, the install rules for `bin` must be setup before anything else. We introduced this in #1354.

Reply:
Oh yeah, I remember now. It's quite unfortunate that this is a bit fragile. Do you have a test case for this as well, or shall I add one?

Reply:
I don't have one, adding one is a good idea.